src/collectives/device/all_reduce.h

```diff
@@ -193,13 +206,19 @@ namespace {
       nthreadsSplit = (nthreads*7/(10*WARP_SIZE))*WARP_SIZE;
     }
 
+    NPKIT_GPU_TREE_SPLIT_DECL_CTX_ID_AND_THREAD_FLAG()
+    NPKIT_GPU_SYNC_TIME(npKitCtxIdx, isNpKitThread)
```

Comment:
since npKitCtxIdx is not defined here, maybe we can hide it inside the function itself.

Reply:
the same comment for the others